lib/Ast.ml

```diff
@@ -158,7 +158,8 @@ module Indexing_op = struct
     | Some (lhs :: args) -> (
       match ident with
       | {pexp_desc= Pexp_ident {txt= ident; loc}; pexp_attributes= []; _}
-        -> (
+      (* We only use the sugared form if it was already used in the source. *)
+        when loc.loc_ghost -> (
```

Comment:
In what cases does `get_sugar_ident` returns None now?

Reply:
In the case `ident` is not an indexing operator (`Array.get`, `.()`, etc..) or the number of argument is not right (if there is not enough or too many, it cannot be a "sugar" expression).

Reply:
That didn't change. `get_sugar` changed however, it returns None in case of labelled arguments, attributes and now there is this new check.